Content.Server/Connection/ConnectionManager.cs

```diff
@@ -31,15 +36,42 @@ public sealed class ConnectionManager : IConnectionManager
         [Dependency] private readonly IConfigurationManager _cfg = default!;
         [Dependency] private readonly ILocalizationManager _loc = default!;
         [Dependency] private readonly ServerDbEntryManager _serverDbEntry = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+        private ISawmill _sawmill = default!;
+        private PlayerConnectionWhitelistPrototype[]? _whitelists;
+        private readonly Dictionary<NetUserId, List<IAdminRemarksRecord>> _adminRemarksCache = new();
 
         public void Initialize()
         {
             _netMgr.Connecting += NetMgrOnConnecting;
             _netMgr.AssignUserIdCallback = AssignUserIdCallback;
+            _sawmill = Logger.GetSawmill("connmgr");
             // Approval-based IP bans disabled because they don't play well with Happy Eyeballs.
             // _netMgr.HandleApprovalCallback = HandleApproval;
         }
 
+        private void UpdateWhitelists(string s)
+        {
+            var list = new List<PlayerConnectionWhitelistPrototype>();
+            var allWhitelists = _prototypeManager.EnumeratePrototypes<PlayerConnectionWhitelistPrototype>().ToList();
+            foreach (var id in s.Split(','))
+            {
+                if (allWhitelists.FirstOrDefault(p => p.ID == id) is { } prototype)
+                {
+                    list.Add(prototype);
+                }
+                else
+                {
+                    _sawmill.Error($"Whitelist prototype {id} does not exist.");
+                    _whitelists = Array.Empty<PlayerConnectionWhitelistPrototype>();
```

Comment:
You are still causing the whitelist system to fail closed (people can connect) on misconfiguration. This should not be a thing.